ako-gateway-api/k8s/ako_init.go

```diff
@@ -1 +1,409 @@
+
+	graphQueue.SyncFunc = SyncFromNodesLayer
+	graphQueue.Run(stopCh, graphwg)
+
+	c.SetupEventHandlers(informers)
+	c.SetupGatewayApiEventHandlers(numWorkers)
+
+	if ctrlAuthToken, ok := utils.SharedCtrlProp().AviCacheGet(utils.ENV_CTRL_AUTHTOKEN); ok && ctrlAuthToken != nil && ctrlAuthToken.(string) != "" {
+		c.RefreshAuthToken()
+	}
+	if ctrlAuthToken, ok := utils.SharedCtrlProp().AviCacheGet(utils.ENV_CTRL_AUTHTOKEN); ok && ctrlAuthToken != nil && ctrlAuthToken.(string) != "" {
+		tokenWorker = utils.NewFullSyncThread(time.Duration(utils.RefreshAuthTokenInterval) * time.Hour)
+		tokenWorker.SyncFunction = c.RefreshAuthToken
+		go tokenWorker.Run()
+	}
+	if lib.DisableSync {
+		gwlib.AKOControlConfig().PodEventf(corev1.EventTypeNormal, lib.AKODeleteConfigSet, "AKO is in disable sync state")
+	} else {
+		gwlib.AKOControlConfig().PodEventf(corev1.EventTypeNormal, lib.AKOReady, "AKO is now listening for Object updates in the cluster")
+	}
+
+	ingestionQueue := utils.SharedWorkQueue().GetQueueByName(utils.ObjectIngestionLayer)
+	ingestionQueue.SyncFunc = SyncFromIngestionLayer
+	ingestionQueue.Run(stopCh, ingestionwg)
+
+	fastRetryQueue := utils.SharedWorkQueue().GetQueueByName(lib.FAST_RETRY_LAYER)
+	fastRetryQueue.SyncFunc = SyncFromFastRetryLayer
+	fastRetryQueue.Run(stopCh, fastretrywg)
+
+	slowRetryQueue := utils.SharedWorkQueue().GetQueueByName(lib.SLOW_RETRY_LAYER)
+	slowRetryQueue.SyncFunc = SyncFromSlowRetryLayer
+	slowRetryQueue.Run(stopCh, slowretrywg)
+
+	statusQueue := utils.SharedWorkQueue().GetQueueByName(utils.StatusQueue)
+	statusQueue.SyncFunc = SyncFromStatusQueue
+	statusQueue.Run(stopCh, statusWG)
+
+LABEL:
+	for {
+		select {
+		case <-quickSyncCh:
+			worker.QuickSync()
+		case <-ctrlCh:
+			break LABEL
+		}
+	}
+	if worker != nil {
+		worker.Shutdown()
+	}
+
+	cancel()
+
+	// Cancel the Leader election goroutines
+	<-ctx.Done()
+
+	ingestionQueue.StopWorkers(stopCh)
+	graphQueue.StopWorkers(stopCh)
+	fastRetryQueue.StopWorkers(stopCh)
+	slowRetryQueue.StopWorkers(stopCh)
+	statusQueue.StopWorkers(stopCh)
+}
+
+func (c *GatewayController) addIndexers() {
+
+	c.informers.ServiceInformer.Informer().AddIndexers(
+		cache.Indexers{
+			lib.AviSettingServicesIndex: func(obj interface{}) ([]string, error) {
+				service, ok := obj.(*corev1.Service)
+				if !ok {
+					return []string{}, nil
+				}
+				if service.Spec.Type == corev1.ServiceTypeLoadBalancer {
+					if val, ok := service.Annotations[lib.InfraSettingNameAnnotation]; ok && val != "" {
+						return []string{val}, nil
+					}
+				}
+				return []string{}, nil
+			},
+		},
+	)
+
+	gwinformer := gwlib.AKOControlConfig().GatewayApiInformers()
+	gwinformer.GatewayInformer.Informer().AddIndexers(
+		cache.Indexers{
+			lib.GatewayClassGatewayIndex: func(obj interface{}) ([]string, error) {
+				gw, ok := obj.(*gwApi.Gateway)
+				if !ok {
+					return []string{}, nil
+				}
+				return []string{string(gw.Spec.GatewayClassName)}, nil
+			},
+		},
+	)
+}
```

Comment:
shall we move this to a different file where we can keep all the indexer related functions?